torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -2962,14 +2961,9 @@ def _post_backward_hook(self, param: Parameter, *unused: Any) -> None:
                     num_pad = self.world_size * chunks[0].numel() - grad.numel()
                     input_flattened = F.pad(grad_flatten, [0, num_pad])
                     output = torch.zeros_like(chunks[0])
-                    dist._reduce_scatter_base(
-                        output, input_flattened, group=self.process_group
-                    )
-                    if self.gradient_postdivide_factor > 1:
-                        # Average grad by pre-division factor. Together pre- and post-division factors
-                        # lead to an overall averaging by world_size, required for consistency with PyTorch DDP.
-                        # This is a two-step process to avoid potential underflow and overflow.
-                        output.div_(self.gradient_postdivide_factor)
+                    # if a communication hook was not registered,
```

Comment:
I am not a proponent of having this comment here. IMHO, this should be documented when `self.communication_hook` is set to the default and only there.